docs/Breadcrumb.md

```diff
@@ -314,12 +317,156 @@ Then, it's the job of the `<Breadcrumb>` component to render the breadcrumb path
+- `/artists/:id/songs/:songId`: `artists.edit.songs.edit` (defined manually)
+
+Let's see how the components for the songs list and detail pages define their app location:
+
+{% raw %}
+```jsx
+// in src/songs/SongList.js
+import { useGetOne, List; SearchInput, Datagrid, TextField, DateField } from 'react-admin';
+import { useDefineAppLocation } from '@react-admin/ra-navigation';
+import { useParams } from 'react-router-dom';
+
+export const SongList = () => {
+    const { id } = useParams();
+    const { data: record } = useGetOne('artists', { id });
+    useDefineAppLocation('artists.edit.songs', { record });
+    return (
+        <List
+            resource="songs"
+            filter={{ artist_id: id }}
+            filters={[<SearchInput key="q" source="q" alwaysOn />]}
+        >
+            <Datagrid>
+                <TextField source="title" />
+                <DateField source="released" />
+                <TextField source="writer" />
+                <TextField source="producer" />
+                <TextField source="recordCompany" label="Label" />
+                <EditSongButton />
+            </Datagrid>
+        </List>
+    );
+};
+
+const EditSongButton = () => {
+    const song = useRecordContext();
+    return (
+        <Button
+            component={Link}
+            to={`/artists/${song?.artist_id}/songs/${song?.id}`}
+            startIcon={<EditIcon />}
+        >
+            Edit
+        </Button>
+    );
+};
+```
+
+```jsx
+// in src/songs/SongDetail.js
+import { useGetOne, Edit, SimpleForm, TextInput, DateInput } from 'react-admin';
+import { useDefineAppLocation } from '@react-admin/ra-navigation';
+import { useParams } from 'react-router-dom';
+
+export const SongDetail = () => {
+    const { id, songId } = useParams();
+    const { data: record } = useGetOne('artists', { id });
+    const { data: song } = useGetOne('songs', { id: songId });
+    useDefineAppLocation('artists.edit.songs.edit', { record, song });
+    return (
+        <Edit resource="songs" id={songId} redirect={`/artists/${id}/songs`}>
+            <SimpleForm>
+                <TextInput source="title" />
+                <DateInput source="released" />
+                <TextInput source="writer" />
+                <TextInput source="producer" />
+                <TextInput source="recordCompany" label="Label" />
+            </SimpleForm>
+        </Edit>
+    );
+};
+```
+{% endraw %}
+
+**Tip**: The `<Edit>` component will call `dataProvider.getOne("songs", { id: songId })` to fetch the song record. Since the `<SongDetail>` component makes the same request, React-admin will deduplicate the calls and only make one request to the dataProvider.
+
+**Tip**: You don't need to call `useDefineAppLocation` for the pages of the parent resource (artists) as this resource uses the default URLs (`/artists`, `/artists/:id/edit`). The app location will be deduced automatically from these URLs (i.e. `artists` and `artists.edit`).
```

Comment:
```suggestion
**Tip**: You don't need to call `useDefineAppLocation` for the pages of the parent resource (artists) as this resource uses the default URLs (`/artists`, `/artists/:id`). The app location will be deduced automatically from these URLs (i.e. `artists` and `artists.edit`).
```

Careful with that _Copilot_ thing of yours :stuck_out_tongue_closed_eyes:

Reply:
This one is a pure human error.